### ISSUE TITLE
[Bug]: [24.5rc0] Simulations with `pybamm.Experiment` terminate after 24 h

### ISSUE THREAD

**Issue by ejfdickinson** [pybamm-team/PyBaMM#4224]
### PyBaMM Version

24.5rc0

### Python Version

3.9.13

### Describe the bug

Simulations using `pybamm.Experiment` with a voltage cut-off only appear to terminate with a time cut-off after 24 h.

I'm afraid I haven't had time to get into logs to figure out why this is happening!

### Steps to Reproduce

The following simulation should proceed for approximately 100 h = 360000 s, until the specified voltage cut-off. However, it terminates after 24 h.

```python
import pybamm

parameter_values = pybamm.ParameterValues("Chen2020")
model = pybamm.lithium_ion.DFN()
experiment = pybamm.Experiment(
    [
        "Discharge at 0.01C until 2.5 V",
    ]
)

sim = pybamm.Simulation(
    model,
    parameter_values=parameter_values,
    experiment=experiment,
)

sol = sim.solve()
sim.plot()

print(f'Last time step at {sim.solution["Time [s]"].entries[-1] / 3600} h')
```

**Comment by agriyakhetarpal** [comment]
Hi, @ejfdickinson – looks like a duplicate of https://github.com/pybamm-team/PyBaMM/issues/4155?

**Comment by ejfdickinson** [comment]
@agriyakhetarpal Yes, it is.

v24.1 does not do this, so it should be in your release notes for v24.5rc0.

Any recommendations on how to work around?

**Comment by MarcBerliner** [comment]
> Any recommendations on how to work around?

@ejfdickinson you can include a duration in the experiment step, e.g.,
```python
experiment = pybamm.Experiment(
    [
        "Discharge at 0.01C for 100 hours or until 2.5 V",
    ]
)
```
and it won't cut off at 10 hours.

<img width="1484" alt="image" src="https://github.com/pybamm-team/PyBaMM/assets/34451391/cb30fa0d-f333-464d-951a-4dc644437e0a">

**Comment by ejfdickinson** [comment]
Thanks @MarcBerliner !

**Comment by agriyakhetarpal** [comment]
> v24.1 does not do this, so it should be in your release notes for v24.5rc0.

I think this we should mark it as a breaking change, @valentinsulzer?

**Comment by agriyakhetarpal** [comment]
> v24.1 does not do this, so it should be in your release notes for v24.5rc0.

I think we should mark this as a breaking change, @valentinsulzer?

**Comment by rtimms** [comment]
IMO we should fix this before 24.5 proper. I don't know what logic changed or if there was a longer default before.

**Comment by rtimms** [comment]
Alternatively, should we require a duration to avoid unexpected behaviour?

**Comment by brosaplanella** [comment]
Why did we move away from that logic? As in, if we bring it back will it break anything?

**Comment by rtimms** [comment]
It got moved to the `Step` class instead of the `Simulation` class. It's easy to put the same logic back in for C-rate, but not current, as the default duration needs to return a float, not a symbol (see the commit above). It's fixable but needs a bit of work, and I'm leaning towards requiring a duration as it's less ambiguous. It would be a breaking change but easy for users to fix.

**Comment by ejfdickinson** [comment]
@rtimms For our use cases, a maximum duration would be fine, so long as it's declared on simulation exit like a `"Time"` stop rather than occurring silently. As noted in #4155 , 24 h is well within typical simulation times for normal experiments, so if the purpose is just to avoid hanging the solver due to an absurd duration, adding a few orders of magnitude to the default might be more sensible!

**Comment by valentinsulzer** [comment]
Should be fixed by #4239 . @ejfdickinson and @aabills can you make sure it works for your use case? Then can @kratman or @agriyakhetarpal  cherry pick this into 24.5? Thanks

**Comment by valentinsulzer** [comment]
We'll still need a "reached default time" warning but that is a more complicated fix
